quasar-core/src/main/java/co/paralleluniverse/fibers/suspend/Stacks.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Quasar: lightweight threads and actors for the JVM.
+ * Copyright (c) 2021, R3 Ltd. All rights reserved.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation
+ *
+ *   or (per the licensee's choosing)
+ *
+ * under the terms of the GNU Lesser General Public License version 3.0
+ * as published by the Free Software Foundation.
+ */
+package co.paralleluniverse.fibers.suspend;
+
+import java.util.function.Supplier;
+
+@SuppressWarnings("unused")
+public final class Stacks {
+    private static volatile Supplier<? extends StackOps> GET_STACK;
```

Comment:
I think the memory model says that this doesn't have to be `volatile` if the threads that call it are created after it is set.  So we could remove the `volatile` if we had an "initialise quasar" call that we did somewhere during node start (and it probably works accidentally already without it).   I wonder to what extent we should bother or is this code riddled with memory barriers anyway?  I guess a suspension point is expensive anyway, so perhaps no need to bother.

Reply:
I was preferring the code to be correct in its own right rather than relying on having "magic" code in a client. However, on the subject of performance, I have found [this article](https://dzone.com/articles/invoke-interface-optimisations) that suggests `invokeinterface` can be slower than `invokevirtual` - although only the method look-up should still be very simple.

However, if you promise _not_ to tell @mnesbit then I can recreate `StackOps` as an `abstract` base class so that the `invokeinterface` instructions can become `invokevirtual` again.

Reply:
I was going to say, is `StackOps` as abstract possible?  Could also then undo java version 8 stuff.  It can have the static `StackOps.getStack()` too, then ? And use `protected` for the static so that `Stack` can initialise?

Reply:
Most of that is easy, but I don't believe having
```java
protected static volatile Supplier<? extends StackOps> GET_STACK;
```
is a good idea. If `Stack` can modify it without using reflection then so can
```java
public class EvilStack extends StackOps {}
```
Remember that `StackOps` _**must**_ be`public`, non-`final`. _**and**_ globally available throughout the OSGi framework! :scream:

Reply:
I think there are mechanisms to eliminate the `volatile` since `MethodHandles` are immutable, but we go down a bit of a rabbit hole of edge cases that probably isn't worth it unless we really run into issues with the `volatile` in the future when looking at performance.